src/traces/indicator/plot.js

```diff
@@ -0,0 +1,788 @@
+    if(trace._hasNumber) bignumberFontSize = trace.number.font.size;
+    if(trace._hasDelta) deltaFontSize = trace.delta.font.size;
+
+    // Position delta relative to bignumber
+    var deltaDy = 0;
+    var deltaX = 0;
+    var bignumberY = 0;
+
+    if(trace._hasDelta && trace._hasNumber) {
+        if(trace.delta.position === 'bottom') {
+            deltaDy = deltaFontSize * 1.5;
+        }
+        if(trace.delta.position === 'top') {
+            deltaDy = -bignumberFontSize + MID_SHIFT * deltaFontSize;
+        }
+        if(trace.delta.position === 'right') {
+            deltaX = undefined;
+        }
+        if(trace.delta.position === 'left') {
+            deltaX = undefined;
+            bignumberY = MID_SHIFT * bignumberFontSize / 2;
+        }
+    }
+    deltaDy -= MID_SHIFT * deltaFontSize;
+
+    var numbers = plotGroup.selectAll('text.numbers').data(cd);
+    numbers.enter().append('text').classed('numbers', true);
+
+    var data = [];
+    if(trace._hasNumber) data.push('number');
+    if(trace._hasDelta) {
+        data.push('delta');
+        if(trace.delta.position === 'left') data.reverse();
+    }
+    var sel = numbers.selectAll('tspan').data(data);
+    sel.enter().append('tspan');
+    sel
+        .attr('text-anchor', function() {return numbersAnchor;})
+        .attr('class', function(d) { return d;})
+        .attr('dx', function(d, i) {
+            // Add padding to the second tspan when it's a one-liner
+            if(i === 1) {
+                var pos = trace.delta.position;
+                if(pos === 'left' || pos === 'right') return 10;
+            }
+            return null;
+        });
+    sel.exit().remove();
+
+    function drawBignumber() {
+        // bignumber
+        var bignumberAx = mockAxis(gd, {tickformat: trace.number.valueformat});
+        var fmt = function(v) { return Axes.tickText(bignumberAx, v).text;};
+        var bignumberSuffix = trace.number.suffix;
+        if(bignumberSuffix) bignumberSuffix = ' ' + bignumberSuffix;
+
+        var number = numbers.select('tspan.number');
+        number
+            .call(Drawing.font, trace.number.font)
+            .attr('x', null)
+            .attr('dy', bignumberY);
+
+        if(hasTransition) {
+            number
+                .transition()
+                .duration(transitionOpts.duration)
+                .ease(transitionOpts.easing)
+                .each('end', function() { trace._lastValue = cd[0].y; onComplete && onComplete(); })
+                .each('interrupt', function() { onComplete && onComplete(); })
+                .attrTween('text', function() {
+                    var that = d3.select(this);
+                    var interpolator = d3.interpolateNumber(cd[0].lastY, cd[0].y);
+                    return function(t) {
+                        that.text(fmt(interpolator(t)) + bignumberSuffix);
+                    };
+                });
+        } else {
+            number.text(fmt(cd[0].y) + bignumberSuffix);
```

Comment:
I was expecting `Lib.constrain(cd[0].y, vmin, vmax)` here.

Does `vmin` and `vmax` only have an effect in for `mode` values that include `'gauge'`?

Reply:
> Does vmin and vmax only have an effect in for mode values that include 'gauge'?

Initially no, but right now, it's only used for setting the range of the `gauge`. I guess we could also constrain the `number` :thinking:

Reply:
... or alternatively make move `vmin` and `vmax` to the `gauge.*` attribute container

Reply:
Merged with https://github.com/plotly/plotly.js/pull/3978#pullrequestreview-254121221